crates/ruff/src/rules/perflint/rules/incorrect_dict_iterator.rs

```diff
@@ -0,0 +1,262 @@
+use ruff_text_size::TextRange;
+use rustpython_parser::ast::Expr;
+
+use crate::registry::AsRule;
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::prelude::Ranged;
+use rustpython_parser::ast;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks if a call to `dict.items()` uses only the keys or values
+///
+/// ## Why is this bad?
+/// Python dictionaries store keys and values in two separate tables. They can be individually
+/// iterated. Using .items() and discarding either the key or the value using _ is inefficient,
+/// when .keys() or .values() can be used instead:
+///
+/// ## Example
+/// ```python
+/// some_dict = {"a": 1, "b": 2}
+/// for _, val in some_dict.items():
+///     print(val)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// some_dict = {"a": 1, "b": 2}
+/// for val in some_dict.values():
+///     print(val)
+/// ```
+#[violation]
+pub struct IncorrectDictIterator {
+    subset: String,
+}
+
+impl AlwaysAutofixableViolation for IncorrectDictIterator {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let IncorrectDictIterator { subset } = self;
+        format!("When using only the {subset} of a dict use the `{subset}()` method")
+    }
+
+    fn autofix_title(&self) -> String {
+        let IncorrectDictIterator { subset } = self;
+        format!("Replace `.items()` with `.{subset}()`.")
+    }
+}
+
+fn is_not_ignored_tuple_or_name(expr: &Expr) -> bool {
+    if let Expr::Name(ast::ExprName { id, .. }) = expr {
+        return id != "_";
+    }
+
+    if let Expr::Tuple(ast::ExprTuple { elts, .. }) = expr {
+        for el in elts {
+            if let Expr::Name(ast::ExprName { id, .. }) = el {
+                // If one of the items of the tuple is not ignored, we consider the whole tuple
+                // as unignored
+                if id != "_" {
+                    return true;
+                }
+                continue;
+            }
+            if let Expr::Tuple(_) = el {
+                return is_not_ignored_tuple_or_name(el);
+            }
+        }
+    }
+    false
+}
+
+/// W8102
+pub(crate) fn incorrect_dict_iterator(checker: &mut Checker, target: &Expr, iter: &Expr) {
+    if let Expr::Call(ast::ExprCall { func, args, .. }) = iter {
```

Comment:
Good suggestion! Done it for a bunch of `if let`'s now. Could go one or two steps further (and now maybe refactoring to some `fn` that can be called on both elements instead of having 4 branches of code for both scenarios for both elements). But already liking its readability much more so your call AFAIAC